### PR TITLE
Imp/jtt 113 update env attributes on run time

### PR DESCRIPTION
**Pull request by ikaratass**
will allow us to update secs settings on run time.

### CODE REVIEW

File: iso15118/secc/comm_session_handler.py [hunk 1]
```diff
@@ -232,6 +232,17 @@ async def __init__.
 
         await wait_for_tasks(self.list_of_tasks)
 
+    def update_config(self, new_config: dict):
+        for key, value in new_config.items():
+            logger.info(
+                f"SECC Config is updated key = {key}: old value = "
+                f"{self.config.as_dict()[key]} - new value = {value}"
+            )
+        self.config.update(new_config)
```
Comment (shalinnijel2) [inline iso15118/secc/comm_session_handler.py:241]: Would it be better to call update() before logging the "updated" message?

File: tests/shared/config/test_config.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    @pytest.mark.parametrize(
+        "config_name, new_value",
+        [
+            ("iface", "en0"),
+            ("log_level", logging.INFO),
+            ("enforce_tls", True),
+            ("free_charging_service", True),
+            ("free_cert_install_service", False),
+            ("use_cpo_backend", True),
+            (
```
Comment (shalinnijel2) [inline tests/shared/config/test_config.py:34]: Would our changes work for shared settings as well? Like PKI_PATH, MESSAGE_LOG_JSON etc?

**Review by shalinnijel2 — CHANGES_REQUESTED** [review]
Hi Ibrahim - I wonder how this

File: tests/shared/config/test_config.py (hunk 2 above)
Comment (ikaratass) [reply tests/shared/config/test_config.py:36]: Done

File: iso15118/secc/comm_session_handler.py (hunk 1 above)
Comment (ikaratass) [reply iso15118/secc/comm_session_handler.py]: Done

File: iso15118/shared/security.py [hunk 3]
```diff
@@ -1449,34 +1450,39 @@ class CertPath(str, Enum):
...
+    OEM_LEAF_DER = "oemLeafCert.der"
+    OEM_SUB_CA2_DER = "oemSubCA2Cert.der"
+    OEM_SUB_CA1_DER = "oemSubCA1Cert.der"
+    OEM_ROOT_DER = "oemRootCACert.der"
+    OEM_ROOT_PEM = "oemRootCACert.pem"
+    OEM_CERT_CHAIN_PEM = "oemCertChain.pem"
+
+    def __get__(self, instance, owner):
+        return os.path.join(
+            shared_settings[SettingKey.PKI_PATH], "iso15118_2/certs/", super().value
```
Comment (shalinnijel2) [inline iso15118/shared/security.py:1484]: Wouldn't self.value work here?

File: iso15118/shared/security.py [hunk 4]
```diff
@@ -1489,30 +1495,35 @@ class KeyPath(str, Enum):
...
+    OEM_LEAF_PEM = "oemLeaf.key"
+    OEM_SUB_CA2_PEM = "oemSubCA2.key"
+    OEM_SUB_CA1_PEM = "oemSubCA1.key"
+    OEM_ROOT_PEM = "oemRootCA.key"
+
+    def __get__(self, instance, owner):
+        return os.path.join(
+            shared_settings[SettingKey.PKI_PATH],
+            "iso15118_2/private_keys/",
+            super().value,
```
Comment (shalinnijel2) [inline iso15118/shared/security.py:1525]: Same comment as above, would self.value be better?

File: iso15118/shared/security.py [hunk 5]
```diff
@@ -1524,18 +1535,15 @@ class KeyPasswordPath(str, Enum):
...
+    OEM_LEAF_KEY_PASSWORD = "oemLeafPassword.txt"
+    CONTRACT_LEAF_KEY_PASSWORD = "contractLeafPassword.txt"
+    CPS_LEAF_KEY_PASSWORD = "cpsLeafPassword.txt"
+    MO_SUB_CA2_PASSWORD = "moSubCA2LeafPassword.txt"
+
+    def __get__(self, instance, owner):
+        return os.path.join(
+            shared_settings[SettingKey.PKI_PATH],
+            "iso15118_2/private_keys/",
+            super().value,
```
Comment (shalinnijel2) [inline iso15118/shared/security.py:1548]: Same as above regarding super().value

File: iso15118/shared/settings.py [hunk 6]
```diff
@@ -2,6 +2,15 @@
 
 import environs
 
+
+class SettingKey:
+    PKI_PATH = "pki_path"
+    MESSAGE_LOG_JSON = "message_log_json"
+    MESSAGE_LOG_EXI = "message_log_exi"
+    V20_SERVICE_CONFIG = "v20_service_config"
```
Comment (shalinnijel2) [inline iso15118/shared/settings.py:10]: If the constants are in upper case, then logging becomes consistent with what we have today. When I ran secc the env were logged like this:
```
DEBUG    2023-10-03 21:47:41,980 - asyncio (54): Using selector: KqueueSelector
INFO    2023-10-03 21:47:41,981 - iso15118.secc.secc_settings (107): SECC settings:
INFO    2023-10-03 21:47:41,981 - iso15118.secc.secc_settings (109): pki_path                      : /Users/shalinnijel/Desktop/Switch/repos/josev_pro_repos/josev_pro_release/iso15118_service/iso15118_service/pki/
INFO    2023-10-03 21:47:41,981 - iso15118.secc.secc_settings (109): message_log_json              : True
INFO    2023-10-03 21:47:41,981 - iso15118.secc.secc_settings (109): message_log_exi               : False
INFO    2023-10-03 21:47:41,981 - iso15118.secc.secc_settings (109): v20_service_config            : /Users/shalinnijel/Desktop/Switch/repos/iso15118_repos/iso15118_release_test/iso15118/shared/examples/secc/15118_20/service_config.json
INFO    2023-10-03 21:47:41,981 - iso15118.secc.secc_settings (109): enable_tls_1_3                : False
INFO    2023-10-03 21:47:41,982 - iso15118.secc.secc_settings (111): NETWORK_INTERFACE             : en0
INFO    2023-10-03 21:47:41,982 - iso15118.secc.secc_settings (111): LOG_LEVEL                     : DEBUG
```

**Review by shalinnijel2 — CHANGES_REQUESTED** [review]

File: iso15118/shared/settings.py (hunk 6 above)
Comment (shalinnijel2) [reply iso15118/shared/settings.py:10]: I see that in secc_settings, while logging, the env variables are automatically picked up - but here, we could still define the constants in upper case and that would fix the logging, right? Anything else that might cause issues if we make them lowercase?

File: iso15118/shared/security.py (hunk 3 above)
Comment (ikaratass) [reply iso15118/shared/security.py]: it worked. updated

File: iso15118/shared/security.py (hunk 5 above)
Comment (ikaratass) [reply iso15118/shared/security.py]: it worked. updated

File: iso15118/secc/secc_settings.py [hunk 7]
```diff
@@ -16,7 +16,7 @@
 @dataclass
 class Config:
     iface: Optional[str] = None
-    log_level: Optional[int] = None
+    log_level: Union[str, int] = None
```
Comment (shalinnijel2) [inline iso15118/secc/secc_settings.py:19]: As discussed, this could be changed to Optional[str]

**Review by shalinnijel2 — CHANGES_REQUESTED** [review]

File: iso15118/shared/settings.py (hunk 6 above)
Comment (ikaratass) [reply iso15118/shared/settings.py]: Done

**Review by shalinnijel2 — APPROVED** [review]

**Pull request merged**